### PR TITLE
Feature/splitimage

### PR DESCRIPTION
**Pull request by martinfalk**
new processor with an adjustable vertical/horizontal image split.

image

### CODE REVIEW

**Review by petersteneteg — APPROVED** [review]

File: modules/basegl/processors/splitimage.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    addProperty(splitDirection_);
+    addProperty(splitPosition_);
+}
+
+void SplitImage::process() {
+    utilgl::activateAndClearTarget(outport_);
+
+    if (!inport0_.hasData() && !inport1_.hasData()) {
+        // no inputs, render noise
```
Comment (ResearchDaniel) [inline modules/basegl/processors/splitimage.cpp:76]: Would be possible to require one of them to be connected using custom isReady

File: modules/basegl/processors/splitimage.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    } else {
+        utilgl::DepthMaskState depthMask(GL_FALSE);
+        utilgl::GlBoolState depth(GL_DEPTH_TEST, GL_FALSE);
+        utilgl::GlBoolState scissor(GL_SCISSOR_TEST, GL_TRUE);
+
+        const ivec2 dims(outport_.getDimensions());
+
+        ivec4 viewport0;
+        ivec4 viewport1;
+        if (splitDirection_.get() == SplitDirection::Vertical) {
```
Comment (ResearchDaniel) [inline modules/basegl/processors/splitimage.cpp:91]: ImageLayoutGL has logic for splitting as well. 
Could perhaps be adopted in future PR...

File: modules/basegl/processors/splitimage.h [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/properties/ordinalproperty.h>
+#include <inviwo/core/properties/optionproperty.h>
+#include <inviwo/core/ports/imageport.h>
+
+namespace inviwo {
+
+/** \docpage{org.inviwo.SplitImage, Split Image}
+ * ![](org.inviwo.SplitImage.png?classIdentifier=org.inviwo.SplitImage)
+ * Split screen of two input images. The images are split in the middle either horizontally or
```
Comment (ResearchDaniel) [inline modules/basegl/processors/splitimage.h:44]: In the middle? Determined by split position

**Review by ResearchDaniel — APPROVED** [review]
Nice!
Potential for common code with ImageLayoutGL.

File: modules/basegl/processors/splitimage.cpp (hunk 1 above)
Comment (martinfalk) [reply modules/basegl/processors/splitimage.cpp:76]: potentially yes, but I thought that not requiring any inputs at all might also be an option.

**Pull request merged**